app/routes.py

```diff
@@ -1,2 +1,73 @@
-from flask import Blueprint
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.models.planet import Planet
+from app import db
 
+planets_bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+@planets_bp.route("", methods=["POST"])
+def create_planet():
+    request_body = request.get_json()
+
+    new_planet = Planet.from_dict(request_body)
+
+    db.session.add(new_planet)
+    db.session.commit()
+
+    return jsonify_message(f"Planet {new_planet.name} successfully created", 201)
+
+@planets_bp.route("", methods=["GET"])
+def read_all_planets():
+    name_query = request.args.get("name")
+    if name_query:
+        planets = Planet.query.filter_by(name=name_query)
+    else: 
+        planets = Planet.query.all() 
+
+    planets_response = [planet.to_dict() for planet in planets]
+    return jsonify(planets_response)
+
+
+@planets_bp.route("/<planet_id>", methods = ["GET"])
+def get_one_planet(planet_id):
+    planet = validate_planet(planet_id)
+    return jsonify(planet.to_dict())
+
+
+@planets_bp.route("/<planet_id>", methods = ["PUT"])
+def update_planet(planet_id):
+    request_body = request.get_json()
+    planet = validate_planet(planet_id)
+
+    planet.name = request_body["name"]
+    planet.description = request_body["description"]
+    planet.moons = request_body["moons"]
+
+    db.session.commit()
+    
+    return jsonify_message(f"Planet succesfully updated. {planet.to_dict()}")
```

Comment:
I'd consider returning the planet name in the message like in `create_planet`, or if you want to return all the data for the planet, maybe separating that into a different key/value pair in the response. Placing the dictionary contents inside a message string sounds like it might not be easy to read.